### ISSUE TITLE
F scaling/potential not updated with current epsilon schedule in discrete barycenter

### ISSUE THREAD

**Issue by Ybawan** [ott-jax/ott#705]
**Describe the bug**
In https://github.com/ott-jax/ott/blob/e241ec3d428909c9978e0af9e41cb440e14249dd/src/ott/solvers/linear/discrete_barycenter.py#L149
```python
parallel_update = jax.vmap(
        lambda f, g, marginal, iter: geom.
        update_potential(f, g, jnp.log(marginal), axis=1),
        in_axes=[0, 0, 0, None]
    )
```

Currently, the iteration is not passed into the f potential (and scaling in non-LSE mode) update, causing the default epsilon scheduler to use the target epsilon instead of the current epsilon in the schedule. 

**To Reproduce**
Printing epsilon from https://github.com/ott-jax/ott/blob/e241ec3d428909c9978e0af9e41cb440e14249dd/src/ott/geometry/geometry.py#L376
will show that it uses the target value of epsilon instead of the current scheduled value.

Passing iter into the update function in parallel update fixes the issue. Wanted to double check that it is indeed unintended behavior. I can submit a small PR if it's helpful :)
